### ISSUE TITLE
SRNN's speed could increase more!

### ISSUE THREAD

**Issue by zepingyu0512** [zepingyu0512/srnn#12]
Recently when I read the codes again, I found the "timedistributed" wrapper in keras actually is not computed parallelized, they use a "for loop" when implementing, so the computing over the subsequences are not parallelized. This means SRNN gets the 10+ speed advantage by only saving the back propagation time, without saving the forward propagation time. So it will have faster speed if we do something to change the "for loop" codes, I will try to do this later.

The code of the timedistributed layer is here: https://github.com/tensorflow/tensorflow/blob/r1.11/tensorflow/python/keras/layers/wrappers.py
They use the rnn-based implementation on line 245 (not real rnn, just a trick, see the code carefully will know):
      _, outputs, _ = K.rnn(
          step,
          inputs,
          initial_states=[],
          input_length=input_shape[1],
          unroll=False)
      y = outputs
 The code of K.rnn is here: https://github.com/tensorflow/tensorflow/blob/r1.11/tensorflow/python/keras/backend.py
On line 3135 they use the "for loop" to compute each subsequence.

I think there may be some ways to improve the for loop:
change cuda codes to do matrix-matrix computing over subsequences
put them into batches
